### DISCUSSION TITLE
Issues with Form Attribute Translations in Laravel's validation.php

### DISCUSSION THREAD

**Discussion by radimnemecek** [livewire/livewire#6767] [1. Help]
### Livewire version
3.0.2

### Which PHP version are you using?
PHP 8.2

I decided to use Laravel's localization features in _lang/en/.._ instead of Livewire's 'as' and 'message'. However, when I use the Forms, the translation doesn't work properly. 

I have no idea where the problem could be. I couldn't find anything in the documentation.

### This works (without Form):
 
**create.blade.php**
```<input wire:model.blur="email" type="email" id="email" />```

**Create.php**
```
class Create extends Component
{
    #[Rule('required')]
    public $email = '';
....
```

**lang/en/validation.php**
```
'attributes' => [
    'email' => 'Email Address'
],
```

**Output:** 

`The Email Address field is required.`

### This doesn't work (when using Form):

**create.blade.php**
```<input wire:model.blur="form.email" type="email" id="email" />```

**Create.php**
```
class Create extends Component
{
    public PostForm $form;
....
```

**PostForm.php**
```
class PostForm extends Form
{  
    #[Rule('required')]
    public $email = '';
...
```

**lang/en/validation.php**
```
'attributes' => [
    'email' => 'Email Address'
    'form.email' => 'Email' // this does not work either
],
```

**Output:** 

`The email field is required.`

**Comment by yzPeedro** [comment]
I had the same problem and solved it by putting the validation in the livewire component instead of the form class

```php
// on livewire Compoment

# instead of using
$this->form->validate(); 

# use
$this->validate(
    rules: $this->form->getComponent()->getRules(), 
    attributes: $this->form->attributesTranslations()
);
````

> If you are using the "rules()" method instead of attributes you should put the rules method instead of "getComponent()->getRules()"

<br/>

```php
// on form class

public function attributesTranslations(): array
{
    return [
        'form.name'     => __('validation.attributes.name'),
        'form.document' => __('validation.attributes.document'),
        'form.email'    => __('validation.attributes.email'),
        'form.address'  => __('validation.attributes.address'),
        'form.phone'    => __('validation.attributes.phone')
    ];
}
````

<br />
<hr>

You can also override the validate method of the Form class

```php
// on form class

public function validate(): mixed
{
    return $this->getComponent()->validate(
        rules: $this->getComponent()->getRules(),
        attributes: $this->attributesTranslations()
    );
}
````

```php
// on livewire component

$this->form->validate();
````

I know this is probably not the best possible solution, but while the issue is not resolved this has helped me

**Reply by dakira** [reply]
Thanks for the workaround @yzPeedro. I'll check it out right now. This is still a bug/regression, though.

The docs [state](https://livewire.laravel.com/docs/validation#opting-out-of-localization:~:text=By%20default%2C%20Livewire%20rule%20messages%20and%20attributes%20are%20localized):
> By default, Livewire rule messages and attributes are localized using Laravel's translate helper
This is clearly not the case. For me, only the messages from `/lang/<code>/validation.php` get picked up, not the attributes.
